python/src/file.rs

```diff
@@ -228,9 +231,11 @@ impl LanceFileMetadata {
 
 #[pyclass]
 pub struct LanceFileWriter {
-    inner: Box<FileWriter>,
+    inner: Arc<Mutex<Box<FileWriter>>>,
 }
 
+unsafe impl Sync for LanceFileReader {}
```

Comment:
Why is this needed?  Wouldn't `LanceFileReader` by `Sync` by default if we change all the `&mut self` to `&self`?  If it isn't then why does it need to be?  The code compiles without the `unsafe impl`.